### PR TITLE
Improve AssaultException

### PR DESCRIPTION
**Pull request by F43nd1r**
**What**: 
 - Fix AssaultException arguments other than String
 - Allow exception creation by static factory method

**Why**:
 - Types were available for exception arguments, but apart from strings none were working
 - Superseeds / Closes #289 

**How**: Full rework of AssaultException

**Checklist**:

- [x] Documentation added
      
- [x] Changelog updated
      
- [X] Tests added
      
- [x] Ready to be merged

### CODE REVIEW

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java [hunk 1]
```diff
@@ -1,100 +1,132 @@
...
+  private List<Object> getExceptionArgumentValues()
+      throws ClassNotFoundException, JsonProcessingException {
+    List<Object> list = new ArrayList<>();
+    for (ExceptionArgument argument : arguments) {
+      Class<?> classType = argument.getClassType();
+      String value = argument.getValue();
+      try {
+        list.add(objectMapper.convertValue(value, classType));
+      } catch (IllegalArgumentException e) {
+        list.add(objectMapper.readValue(value, classType));
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java:89]: would it make sense to add some debug log messages in this method? especially for the "exception case"?

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java [hunk 2]
```diff
@@ -1,100 +1,132 @@
...
+    for (ExceptionArgument argument : arguments) {
+      list.add(argument.getClassType());
+    }
+    return list;
+  }
+
+  @JsonIgnore
+  private List<Object> getExceptionArgumentValues()
+      throws ClassNotFoundException, JsonProcessingException {
+    List<Object> list = new ArrayList<>();
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java:82]: lets rename this variable into something like exceptionArguments or something like this (or just values)

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java [hunk 3]
```diff
@@ -1,100 +1,132 @@
...
 
   @JsonIgnore
-  private List<Object> getExceptionArgumentValues() {
-    return arguments.stream().map(ExceptionArgument::getValue).collect(Collectors.toList());
+  public List<Class<?>> getExceptionArgumentTypes() throws ClassNotFoundException {
+    List<Class<?>> list = new ArrayList<>();
+    for (ExceptionArgument argument : arguments) {
+      list.add(argument.getClassType());
+    }
+    return list;
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java:76]:
```suggestion
    List<Class<?>> exceptionArgumentTypes = new ArrayList<>();
    for (ExceptionArgument argument : arguments) {
      exceptionArgumentTypes.add(argument.getClassType());
    }
    return exceptionArgumentTypes;
```

**Review by WtfJoke — COMMENTED** [review]
I havent tested it, but the code looks good. Appreciate the newly added tests and the fix for non string types.


I've left a few comments, but I also miss the documentation update on how to use the new functionality.

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java (hunk 1 above)
Comment (F43nd1r) [reply chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java]: I added some comments. The "exception case" isn't really exceptional, but rather expected if the value is a json formatted string.

**Review by WtfJoke — APPROVED** [review]
LGTM

**Pull request merged**